extensions/resource-deployment/src/interfaces.ts

```diff
@@ -300,9 +300,9 @@ export interface DynamicOptionsInfo {
 }
 
 export interface DynamicOptionsAlternates {
-	selection: string,
-	alternateValues: string[],
-	defaultValue: string
+	selection: string | azdata.CategoryValue,
+	alternateValues: string[] | azdata.CategoryValue[],
+	defaultValue: string | azdata.CategoryValue
```

Comment:
Does the default value here work with a CategoryValue? I can't remember if it's doing a direct comparison or correctly checking the type (and comparing the value). Make sure you validate that. 

In general I don't think this is correct to allow a CategoryValue here though. The value should be the name of the option to use as the default if the options are CategoryValues - having to define the full object again here shouldn't be necessary.